typedoc.json

```diff
@@ -0,0 +1,33 @@
+{
+	"$schema": "https://typedoc.org/schema.json",
+	"categoryOrder": [
+		"Node Types",
+		"Nodes - Type Guards",
+		"Nodes - Flag Utilities",
+		"Nodes - Other Utilities",
+		"Modifier Utilities",
+		"Scope Utilities",
+		"Syntax Utilities",
+		"Compiler Options",
+		"*",
+		"Other"
+	],
+	"cleanOutputDir": true,
+	"defaultCategory": "Other",
+	"entryPointStrategy": "resolve",
+	"entryPoints": ["./src/index.ts"],
+	"exclude": ["./**/test/", "./**/*.test.*"],
+	"excludeExternals": true,
+	"excludeInternal": true,
+	"excludeNotDocumented": false,
+	"excludePrivate": true,
+	"hideGenerator": true,
+	"includeVersion": true,
+	"out": "./docs",
+	"plugin": [
+		"typedoc-plugin-coverage",
+		"typedoc-plugin-mdn-links",
+		"typedoc-plugin-versions"
+	],
+	"readme": "none"
```

Comment:
[Request] Although I agree the full README.md's contents probably aren't what we want for the deployed website, it does feel a little confusing to have ... no introduction at all:

![Screenshot of the typedoc-generated index page with links and no sentence descriptions](https://user-images.githubusercontent.com/3335181/219901473-09c18031-751d-4de6-ac6f-7bf7fb1f97c3.png)

Can we have the section of the README.md that mentions usage and what this package is? I don't see a way to do this other than either [calling TypeDoc programmatically](https://typedoc.org/guides/installation/#node-module) _(eek!)_ or having a dedicated `.md` file. The dedicated file seems best to me?

Reply:
Do you want write that file?

Reply:
Can do! For now I think it's fine to just copy & paste a sentence or three from the README.md. I can take this on as a followup.

Reply:
#145 -> #149